op-program/client/l2/engineapi/block_processor.go

```diff
@@ -94,18 +95,17 @@ func NewBlockProcessorFromHeader(provider BlockDataProvider, h *types.Header) (*
 		vmenv := vm.NewEVM(context, statedb, provider.Config(), vm.Config{PrecompileOverrides: precompileOverrides})
 		return vmenv
 	}
+	vmenv := mkEVM()
```

Comment:
`mkEVM` itself can maybe be improved while we're here: make `provider.GetVMConfig` return a `vm.Config` instead of a `*vm.Config` (a pointer). All implementations already return non-nil configs anyway. And then the config can be passed around by value, and we don't end up with the awkward `vm.Config` reconstruction in `mkEVM`